src/utils/Cache.ts

```diff
@@ -22,27 +22,42 @@ const memoryCache = new Map();
 // By default we will use cache_api if availble, if not we will cache in memory
 // user can also specify to just use in memory caching
 // If user provides a CacheTarget.CACHE_API and cache_api is not availble we will fallback to memory
-export function cacheFactory(optionalTargets: CacheTargets): ShCache {
+export async function cacheFactory(optionalTargets: CacheTargets): Promise<ShCache> {
   const targets = optionalTargets || SUPPORTED_TARGETS;
-  const target = getFirstUseableTarget(targets);
+  const target = await getFirstUseableTarget(targets);
   return constructCache(target);
 }
 
-function getFirstUseableTarget(targets: CacheTargets): CacheTarget {
+async function getFirstUseableTarget(targets: CacheTargets): Promise<CacheTarget> {
   let firstTargetToUse = undefined; // default to memory if target is not supported
   for (const key of targets) {
-    if (doesTargetExist(key)) {
+    if (await doesTargetExist(key)) {
       firstTargetToUse = key;
       break;
     }
   }
   return firstTargetToUse;
 }
 
-export function doesTargetExist(target: CacheTarget): boolean {
+async function checkIfCacheApiAvailable(): Promise<boolean> {
+  if (!window.caches) {
+    return false;
+  }
+  try {
+    const caches = window.caches;
+    //make request to cacheApi to check if it is available
+    await caches.keys();
+    return true;
+  } catch (err) {
+    console.error('CacheApi is not available', err);
```

Comment:
Maybe a warning would be enough here? It is not really an error, it is working as intended.

Also, it would be nice if it was displayed only once, but that means that this function should remember its result (which might be a good idea from performance point of view too), so I'm not sure if this is still within the scope. 

Fftr and merge.